src/linear_algebra/finite_dimensional.lean

```diff
@@ -196,6 +196,162 @@ begin
   exact (lift_inj.mp this).symm
 end
 
+-- Note here we've restrictied the universe levels of `ι` and `V` to be the same, for convenience.
+lemma cardinal_mk_le_findim_of_linear_independent
+  [finite_dimensional K V] {ι : Type v} {b : ι → V} (h : linear_independent K b) :
+  cardinal.mk ι ≤ findim K V :=
+begin
+  have := cardinal_le_dim_of_linear_independent h,
+  rw ←findim_eq_dim K V at this,
+  exact this,
+end
+
+-- Note here we've restrictied the universe levels of `ι` and `V` to be the same, for convenience.
+lemma fintype_card_le_findim_of_linear_independent
+  [finite_dimensional K V] {ι : Type v} [fintype ι] {b : ι → V} (h : linear_independent K b) :
+  fintype.card ι ≤ findim K V :=
+begin
+  have t := cardinal_mk_le_findim_of_linear_independent h,
+  rw fintype_card at t,
+  simpa using t,
+end
+
+lemma finset_card_le_findim_of_linear_independent [finite_dimensional K V] {b : finset V}
+  (h : linear_independent K (λ x, x : (↑b : set V) → V)) :
+  b.card ≤ findim K V :=
+begin
+  rw ←fintype.card_coe,
+  exact fintype_card_le_findim_of_linear_independent h,
+end
+
+section
+open_locale big_operators
+open finset
+/--
+If a finset has cardinality larger than the dimension of the space,
+then there is a nontrivial linear relation amongst its elements.
+-/
+lemma exists_nontrivial_relation_of_dim_lt_card
+  [finite_dimensional K V] {t : finset V} (h : findim K V < t.card) :
+  ∃ f : V → K, ∑ e in t, f e • e = 0 ∧ ∃ x ∈ t, f x ≠ 0 :=
+begin
+  have := mt finset_card_le_findim_of_linear_independent (by { simpa using h }),
+  rw linear_dependent_iff at this,
+  obtain ⟨s, g, sum, z, zm, nonzero⟩ := this,
+  -- Now we have to extend `g` to all of `t`, then to all of `V`.
+  let f : V → K := λ x, if h : x ∈ t then if (⟨x, h⟩ : (↑t : set V)) ∈ s then g ⟨x, h⟩ else 0 else 0,
+  -- and finally clean up the mess caused by the extension.
+  refine ⟨f, _, _⟩,
+  { dsimp [f],
+    convert sum using 1,
+    fapply sum_bij_ne_zero,
+    { intros, exact ⟨a, H⟩, },
+    { intros, dsimp,
+      rw [dif_pos h₁] at h₂,
+      by_contradiction w,
+      rw [if_neg w] at h₂,
+      simpa using h₂, },
+    { intros _ _ _ _ _ _, exact subtype.mk.inj, },
+    { intros, dsimp,
+      use b,
+      have h₁ : ↑b ∈ t := by simp,
+      use h₁,
+      rw [dif_pos h₁, if_pos],
+      { fsplit; simp; assumption, },
+      { simpa, } },
+    { intros a h₁, dsimp, rw [dif_pos h₁], intro h₂,
+      split_ifs with h₃,
+      { refl, },
+      { rw [if_neg h₃, zero_smul, eq_self_iff_true, not_true] at h₂, contradiction, }, }, },
+  { refine ⟨z, z.2, _⟩, dsimp only [f], erw [dif_pos z.2, if_pos]; rwa [subtype.coe_eta] },
+end
+
+/--
+If a finset has cardinality larger than `findim + 1`,
+then there is a nontrivial linear relation amongst its elements,
+such that the coefficients of the relation sum to zero.
+-/
+lemma exists_nontrivial_relation_sum_zero_of_dim_succ_lt_card
```

Comment:
I feel like this deserves some cleaning up.

Reply:
I pushed some further cleanups. I'm still not completely happy with the shape of these proofs. But I don't see immediate further improvements.

Reply:
I added some comments, but didn't change the proof. I have to admit I'm not excited about working on this proof: it's easy verification of new facts from existing facts, so all just shuffling stuff around. I really wish automation helped more here.